test/conftest.py

```diff
@@ -734,3 +777,144 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
         ws_id,
         clients["ws"],
     )
+
+
+# A factory function to generate sets
+def set_factory(
+    set_item_name: str,
+    refs: list[str] | None,
+    set_api_client: SetAPI,
+    context: dict[str, Any],
+    ws_id: int,
+    name_qualifier: str | None = None,
+) -> dict[str, Any]:
+    """Factory function to create a set from a list of parameters.
+
+    :param set_item_name: name of the objects in the set, e.g. "genome"
+    :type set_item_name: str
+    :param refs: list of items to go in the set
+    :type refs: list[str] | None
+    :param set_api_client: the SetAPI client
+    :type set_api_client: SetAPI
+    :param context: KBase context
+    :type context: dict[str, Any]
+    :param ws_id: workspace ID
+    :type ws_id: int
+    :param name_qualifier: extra qualifier to go in the set name, defaults to None
+    :type name_qualifier: str | None, optional
+    :return: output of save_set / save_empty_set
+    :rtype: dict[str, Any]
+    """
+    if not refs:
+        return save_empty_set(set_item_name, set_api_client, context, ws_id)
+
+    return save_set(set_item_name, set_api_client, context, refs, ws_id, name_qualifier)
```

Comment:
Abstracted out the save_set function (into test/util.py) so creating and saving sets now has a user-friendly interface.

Note that they use the ws_id param, which means you can save the set in either the list_all_sets or the default workspace. It also ensures that any fixtures (the `refs` in the `save_set` function) that are used to create this fixture are also created using the appropriate value for the ws_id param. (It's turtles all the way down.)